schemas/iso19139/src/main/plugin/iso19139/process/thumbnail-remove.xsl

```diff
@@ -25,22 +25,37 @@
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
                 xmlns:gmd="http://www.isotc211.org/2005/gmd"
                 xmlns:gco="http://www.isotc211.org/2005/gco"
+                xmlns:xs="http://www.w3.org/2001/XMLSchema"
                 xmlns:geonet="http://www.fao.org/geonetwork" exclude-result-prefixes="#all"
+                xmlns:digestUtils="java:org.apache.commons.codec.digest.DigestUtils"
+                xmlns:exslt="http://exslt.org/common"
+                exclude-result-prefixes="#all"
                 version="2.0">
 
   <!--
       Usage:
+        thumbnail_url is the url to be removed - it is for backwards compatibility.  Will not be used if resourceHash is set.
+        resourceHash is hash value of the object to be removed which will ensure the correct value is removed. It will override the usage of thumbnail_url
+        resourceIdx is the index location of the object to be removed - can be used when duplicate entries exists to ensure the correct one is removed.
+
+      example:
         thumbnail-from-url-remove?thumbnail_url=http://geonetwork.org/thumbnails/image.png
     -->
 
-  <xsl:param name="thumbnail_url"/>
+  <xsl:param name="thumbnail_url" select="''"/>
+  <xsl:param name="resourceHash" select="''"/>
+  <xsl:param name="resourceIdx" select="''"/>
 
   <!-- Remove the thumbnail define in thumbnail_url parameter -->
+  <!-- Note: first part of the match needs to match the xsl:for-each select from extract-relations.xsl in order to get the position() to match -->
```

Comment:
I would write a comment explaining how it works, something like this:

> If provided `resourceHash` and `resourceIdx` matches the thumbnail to delete using that fields. When not provided (backwards compatibility) uses the url to match the thumbnail to delete.

That would facilitate the maintenance in the long term, as the condition is not that simple.

Please if you can also update the other xslt files with similar conditions.

Reply:
I have added little comment there stating the position and how hashing works. Please check